src/python/pants/backend/java/compile/javac_test.py

```diff
@@ -623,10 +623,10 @@ def test_compile_with_maven_deps(rule_runner: RuleRunner) -> None:
     resolved_joda_lockfile = CoursierResolvedLockfile(
         entries=(
             CoursierLockfileEntry(
-                coord=MavenCoord(coord="joda-time:joda-time:2.10.10"),
```

Comment:
Imo these kwarg args aren't really adding much value. Positional arg seems simpler 

Ditto the rest of these files

Reply:
The next refactor PR will split it into `group`, `artifact`, and `version` keys which are useful. I just didn't dio the modification in this rename-only pR.